genai-perf/genai_perf/config/input/config_endpoint.py

```diff
@@ -64,11 +65,162 @@ def parse(self, endpoint: Dict[str, Any]) -> None:
                 self.service_kind = value
             elif key == "streaming":
                 self.streaming = value
-            elif key == "server_metrics_url":
-                self.server_metrics_url = value
+            elif key == "server_metrics_url" or key == "server_metrics_urls":
+                self.server_metrics_urls = self._parse_server_metrics_url(value)
             elif key == "url":
                 self.url = value
             else:
                 raise ValueError(
                     f"User Config: {key} is not a valid endpoint parameter"
                 )
+
+    def _parse_server_metrics_url(self, value: Any) -> List[str]:
+        if type(value) is str:
+            return [value]
+        elif type(value) is list:
+            return value
+        else:
+            raise ValueError(
+                "User Config: server_metrics_url(s) must be a string or list"
+            )
+
+    ###########################################################################
+    # Illegal Combination Methods
+    ###########################################################################
+    def check_for_illegal_combinations(self) -> None:
+        self._check_service_kind_and_type()
+        self._check_server_metrics_url()
+
+    def _check_service_kind_and_type(self) -> None:
+        if not self.type:
+            if (
+                not self.service_kind == "triton"
+                or not self.service_kind == "tensorrtllm_engine"
+            ):
+                raise ValueError(
+                    f"User Config: service_kind {self.service_kind} requires a type to be specified"
+                )
+
+    def _check_server_metrics_url(self) -> None:
+        if self.service_kind == "triton" and self.server_metrics_urls:
+            for url in self.server_metrics_urls:
+                self._check_for_valid_url(url)
+
+    def _check_for_valid_url(self, url: str) -> None:
+        """
+        Validates a URL to ensure it meets the following criteria:
+        - The scheme must be 'http' or 'https'.
+        - The netloc (domain) must be present OR the URL must be a valid localhost
+        address.
+        - The path must contain '/metrics'.
+        - The port must be specified.
+
+        Raises:
+            ValueError if the URL is invalid.
+
+        The URL structure is expected to follow:
+        <scheme>://<netloc>/<path>;<params>?<query>#<fragment>
+        """
+        parsed_url = urlparse(url)
+
+        if parsed_url.scheme not in ["http", "https"]:
+            raise ValueError(
+                f"Invalid scheme '{parsed_url.scheme}' in URL: {url}. Use 'http' "
+                "or 'https'."
+            )
+
+        valid_localhost = parsed_url.hostname in ["localhost", "127.0.0.1"]
```

Comment:
_Technically_, `[::1]` is also a valid localhost value, when using IPv6.

Reply:
My goal here was parity with the checks we currently have in place (in argparse). I'm reluctant to change this check on this branch/in this story.